addon/components/frost-chart-x-axis-svg.js

```diff
@@ -0,0 +1,148 @@
+    label: PropTypes.string,
+    ticks: PropTypes.func,
+    offset: PropTypes.number,
+
+    chartState: PropTypes.EmberObject.isRequired,
+    dispatch: PropTypes.func.isRequired
+
+    // state
+  },
+
+  getDefaultProps () {
+    return {
+      // options
+      aligment: 'bottom',
+      label: null,
+      ticks: linearTicks([10]),
+      offset: 0,
+
+      // state
+      _axis: 'x',
+      // FIXME: Calculate the height of the text to get the height of the svg in the axis
+      _svgHeight: 20
+
+    }
+  },
+
+  // == Computed Properties ===================================================
+
+  @readOnly
+  @computed('chartState.chart.initialized', 'chartState.domain.x')
+  _ticks (initializedChart, domain) {
+    if (!initializedChart) {
+      return []
+    }
+
+    return this.ticks(domain)
+  },
+
+  @readOnly
+  @computed('_ticks', 'chartState.range.x', 'chartState.domain.x')
+  _positionedTicks (ticks, range, domain) {
+    if (!range || !domain) {
+      return ticks
+    }
+
+    const scale = this.get('chartState.scale.x')
+    const transform = scale({domain, range})
+
+    return ticks.map(tick => {
+      return assign({}, tick, {
+        coordinate: transform(get(tick, 'value'))
+      })
+    })
+  },
+
+  @readOnly
+  @computed('chartState.axes.initialized', 'chartState.chart.width')
+  /* eslint complexity: [2, 7] */
+  style (initializedAxes, chartWidth) {
+    if (!initializedAxes || !chartWidth) {
+      return EmberString.htmlSafe('')
+    }
+
+    const chartPadding = this.get('chartState.chart.padding')
+    const yAxisAlignment = this.get('chartState.axes.y.alignment')
+    const yAxisWidth = this.get('chartState.axes.y.width') || 0
+    const xAxisAlignment = this.get('chartState.axes.x.alignment')
+    const xAxisFirstTickMargin = this.get('chartState.axes.x.firstTickMargin')
+    const xAxisLastTickMargin = this.get('chartState.axes.x.lastTickMargin')
+
+    return EmberString.htmlSafe(`
+      ${xAxisAlignment}: ${get(chartPadding, xAxisAlignment)}px;
+      margin-left: calc(${yAxisAlignment === 'left' ? yAxisWidth : xAxisFirstTickMargin}px);
+      margin-right: calc(${yAxisAlignment === 'right' ? yAxisWidth : xAxisLastTickMargin}px);
+    `)
+  },
+
+  @readOnly
+  @computed('chartState.axes.initialized', 'chartState.chart.width')
+  _svgWidth (initializedAxes, chartWidth) {
+    const yAxisWidth = this.get('chartState.axes.y.width') || 0
+    const xAxisFirstTickMargin = this.get('chartState.axes.x.firstTickMargin')
+    const xAxisLastTickMargin = this.get('chartState.axes.x.lastTickMargin')
+
+    return chartWidth - yAxisWidth - xAxisFirstTickMargin - xAxisLastTickMargin
+  },
+
+  // == Functions =============================================================
+
+  _dispatchRenderedAxis () {
+    this.dispatch({
+      type: 'RENDERED_X_AXIS',
+      axis: {
+        alignment: this.get('alignment'),
+        height: this.$().outerHeight(true) + this.get('offset'),
+        ticks: this.get('_ticks'),
```

Comment:
I could be wrong, but I believe we had proven that this wasn't necessary anymore, since we explicitly repeat the tick function in the grid?